test/community/s_metric.jl

```diff
@@ -0,0 +1,22 @@
+using Random, Statistics
+
+@testset "S-metric" begin
+    @testset "Directed ($seed)" for seed in [1, 2, 3], (_n, _ne) in [(14, 18), (10, 22), (7, 16)]
+        g = erdos_renyi(_n, _ne; is_directed=true, seed=seed)
+        sm = s_metric(g, norm=false)
+        sm2 = sum([outdegree(g, src(d)) * indegree(g, dst(d)) for d in edges(g)])
+        @test @inferred sm ≈ sm2
+        sm = s_metric(g, norm=true)
```

Comment:
```suggestion
        @test  sm ≈ sm2
        sm = @inferred s_metric(g, norm=true)
```
The [inferred macro](https://docs.julialang.org/en/v1/stdlib/Test/#Test.@inferred) is used to check if a function is type stable, i.e. the compiler can infer its resulting argument. What you are testing here is, if `≈` is type stable, which is probably not what you want.